qutip/qip/noise.py

```diff
@@ -43,83 +43,29 @@ def process_noise(pulses, noise_list, dims, t1=None, t2=None,
     Returns
     -------
     noisy_pulses: list of :class:`qutip.qip.Pulse`
-        The noisy pulses.
+        The noisy pulses, including the system noise.
     """
-    # first the control pulse with noise,
-    # then additional pulse independent noise.
-    noisy_pulses = process_pulse_noise(pulses, noise_list, dims)
-    if device_noise:
-        noisy_pulses += process_device_noise(noise_list, dims, t1, t2)
-    return noisy_pulses
-
-
-def process_device_noise(noise_list, dims, t1=None, t2=None):
-    """
-    Apply pulse independent noise to the input list of pulses.
-    It does not modify the input
-    pulse, but return a new one containing the noise.
-    Pulse dependent noise will be ignored.
+    if device_noise:  # first pulse independent noise
+        systematic_noise = Pulse(None, None, label="systematic_noise")
+        if (t1 is not None) or (t2 is not None):
+            RelaxationNoise(t1, t2).get_noisy_dynamics(dims, systematic_noise)
```

Comment:
Now I hate OO style overall, but this seems to break its general principles to me - shouldn't all subclasses of `Noise` have compatible call signatures?  Otherwise they're not polymorphic to each other at all, and the `Noise` class isn't correctly defining its interface.  Subclassing is meant to _avoid_ behaviour like
```
if isinstance(x, Subclass1):
    ...
elif isinstance(x, Subclass2):
    ...
elif isinstance(x, Subclass3):
    ...
```
by allowing objects to change their behaviour, but not their interface.

Reply:
Yes, you are right, I should unify the signature of get_noisy_dynamics to be `def get_noisy_dynamics(self, pulses, systematic_noise, dims)`